### DISCUSSION TITLE
Bugs in NTTFriendlyPrimesGenerator

### DISCUSSION THREAD

**Discussion by JoowonLee1209** [tuneinsight/lattigo#496] [Q&A]
Hello,

I found that ring.NewNTTFriendlyPrimesGenerator does not check if NextPrime and PrevPrime are prime numbers in v6.

**Comment by lehugueni** [comment]
Hi!

I think the names are slightly misleading, `NextPrime` and `PrevPrime` are the next values of the form `2^BitSize + k*NThRoot + 1` to be checked for primality. When calling the methods to generate primes (e.g. `NextUpstreamPrime`), we check whether `NextPrime` is actually prime and we increment `NextPrime` by `NThRoot`.
